plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/LibvirtStorageAdaptor.java

```diff
@@ -637,6 +639,39 @@ public KVMPhysicalDisk getPhysicalDisk(String volumeUuid, KVMStoragePool pool) {
         }
     }
 
+    /**
+     * adjust refcount
+     */
+    private int adjustStoragePoolRefCount(String uuid, int adjustment) {
+        synchronized (uuid) {
```

Comment:
Will this synchronized work as expected? Do we guarantee that every String with the same uuid as value is the same object?

Reply:
Yes @JoaoJandre as we use a (concurrent)Hashmap with the uuid as key.

Reply:
typo, i meant retrieve the key from the hashmap and use that as mutex

Reply:
and, yes, it *should* work